test/release/examples/primers/replicated.chpl

```diff
@@ -0,0 +1,169 @@
+// three indices per locale that the distribution is targeting (by
+// default, all locales).  Similarly, an array declared over that
+// domain will store three elements per locale.  Each locale's copy of
+// the domain or array is known as its *replicand*.
+//
+// Consistency among these array replicands is NOT maintained
+// automatically; users who want a replicated array to store the same
+// values on every target locale will have to manage that consistency
+// themselves.
+//
+// In general, operations on replicated domains and arrays only refer
+// the local replicand.  The primary exception to this rule is the
+// re-assignment of a replicated domain's indices.  In this case, the
+// copy of the domain on each locale will be updated (and any arrays
+// over the domain will be reallocated on each locale).
+//
+// Here's a declaration of a replicated domain and array:
+//
+const ReplicatedSpace = Space dmapped Replicated();
+var RA: [ReplicatedSpace] int;
+
+// Queries about the size of a replicated domain or array will return
+// the size per locale:
+//
+writeln("Replicated Array has ", RA.size, " elements per locale");
+
+//
+// The following loop-based assignment to `RA` only affects the copy
+// of 'RA' on the locale on which it's running (the last locale, in
+// this example).  All other copies will remain in their
+// default-initialized form.
+//
+on Locales[numLocales-1] do
+  forall ra in RA do
+    ra = here.id;
+
+//
+// Similarly, when reading the array, only the local copy will be
+// accessed.  Thus, when running on more than one locale, the
+// following writeln() will not see the modification performed by the
+// loop above since the two statements executed on distinct locales:
+//
+writeln("Locale 0's copy of RA is:\n", RA);
+
+
+//
+// To access the replicands owned by other locales, we can use the
+// replicand method, which takes a locale as an argument and return the
+// array local to that locale:
+//
+
+writeln("Locale ", numLocales-1, "'s copy of RA is:\n",
+         RA.replicand(Locales[numLocales-1]));
+
+//
+// Alternatively, we can print the replicated array from that locale:
+//
+
+on Locales[numLocales-1] do
+  writeln("Locale ", numLocales-1, "'s copy of RA is:\n", RA);
+
+//
+// In order to see the replicands owned by all the locales, let's
+// define and call a little utility function:
+//
+proc writeReplicands(X) {
+  for loc in Locales {
+    writeln(loc, ":");
+    writeln(X.replicand(loc));
+  }
+}
+
+writeln("Replicated Array Index Map");
+writeReplicands(RA);
+writeln();
+
+//
+// Whole-array assignment is similarly local only to the current
+// locale's copy of the array:
+//
+var A: [Space] int = [(i,j) in Space] i*100 + j;
+RA = A;
+writeln("Replicated Array after whole-array assignment:");
+writeReplicands(RA);
+writeln();
+
+//
+// Here, we have each locale update its own copy of `RA` to store its
+// locale ID, which results in a modification to each replicand:
+//
+for loc in Locales do
```

Comment:
I think it would be better to keep this as a coforall-on. If this file is intended to demonstrate the recommended usage of ReplicatedDist, I think a coforall-on is more likely than a serial loop using ``replicand``.